<launch>
  <arg name="input_autoware_state" default="/autoware/state" />
  <arg name="input_driving_capability" default="/vehicle/driving_capability" />
  <arg name="input_prev_control_command" default="/vehicle/vehicle_command" />
  <arg name="input_state_report" default="/vehicle/state_report" />
  <arg name="input_odometry" default="/vehicle/odometry" />

  <arg name="output_control_command" default="/vehicle/emergency/vehicle_command" />
  <arg name="output_state_command" default="/vehicle/emergency/state_command" />
  <arg name="output_is_emergency" default="/vehicle/emergency/is_emergency" />
  <arg name="output_hazard_status" default="/vehicle/emergency/hazard_status" />
  <arg name="output_diagnostics_err" default="/diagnostics_err" />

  <arg name="service_clear_emergency" default="/system/clear_emergency" />

  <arg name="config_file" default="$(find-pkg-share emergency_handler)/config/emergency_handler.param.yaml" />
  <arg name="emergency_hazard_level" default="2" description="1: Safe Fault, 2: Latent Fault, 3: Single Point Fault" />
  <arg name="use_emergency_hold" default="false" />

  <!-- emergency_handler -->
  <node pkg="emergency_handler" exec="emergency_handler" name="emergency_handler" output="screen">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>

    <remap from="~/input/autoware_state" to="$(var input_autoware_state)"/>
    <remap from="~/input/driving_capability" to="$(var input_driving_capability)"/>
    <remap from="~/input/prev_control_command" to="$(var input_prev_control_command)"/>
    <remap from="~/input/state_report" to="$(var input_state_report)"/>
    <remap from="~/input/odometry" to="$(var input_odometry)"/>

    <remap from="~/output/control_command" to="$(var output_control_command)"/>
    <remap from="~/output/state_command" to="$(var output_state_command)"/>
    <remap from="~/output/is_emergency" to="$(var output_is_emergency)"/>
    <remap from="~/output/hazard_status" to="$(var output_hazard_status)"/>
    <remap from="~/output/diagnostics_err" to="$(var output_diagnostics_err)"/>

    <remap from="service/clear_emergency" to="$(var service_clear_emergency)"/>

    <param from="$(var config_file)" />
    <param name="emergency_hazard_level" value="$(var emergency_hazard_level)" />
    <param name="use_emergency_hold" value="$(var use_emergency_hold)" />
  </node>
</launch>
